src/main/java/com/booster/server/NoteService.java

```diff
@@ -42,4 +43,13 @@ public List<NoteDto> batchOfSize(Integer size) {
                 .toList();
     }
 
+    @Transactional
+    public void deleteById(Long id) {
+        try {
+            noteRepository.deleteById(id);
+        } catch (EmptyResultDataAccessException e) {
+            throw new EntityNotFoundException(NoteEntity.class, "id", id);
+        }
```

Comment:
EmptyResultDataAccessException has a quite good description. Why do we need to rethrow a  custom exception here?

Reply:
`EmptyResultDataAccessException` is data access later specific and usually has no meaning to the end user.

Also, the message of the original exception is similar to `No class com.booster.server.NoteEntity entity with id 1000 exists!` which is a bit strange. We have no need of exposing the package structure to the end user, etc, so we need a custom exception.

Reply:
Okay